heartbeat/sender/peerAuthenticationSenderFactory.go

```diff
@@ -25,105 +22,53 @@ type argPeerAuthenticationSenderFactory struct {
 	redundancyHandler        heartbeat.NodeRedundancyHandler
 }
 
-type peerAuthenticationSenderFactory struct {
-	messenger                 heartbeat.P2PMessenger
-	marshaller                marshal.Marshalizer
-	topic                     string
-	timeBetweenSends          time.Duration
-	timeBetweenSendsWhenError time.Duration
-	thresholdBetweenSends     float64
-	nodesCoordinator          heartbeat.NodesCoordinator
-	peerSignatureHandler      crypto.PeerSignatureHandler
-	hardforkTrigger           heartbeat.HardforkTrigger
-	hardforkTimeBetweenSends  time.Duration
-	hardforkTriggerPubKey     []byte
-	keysHolder                heartbeat.KeysHolder
-	timeBetweenChecks         time.Duration
-	shardCoordinator          process.ShardCoordinator
-	privKey                   crypto.PrivateKey
-	redundancyHandler         heartbeat.NodeRedundancyHandler
-}
-
-func newPeerAuthenticationSenderFactory(args argPeerAuthenticationSenderFactory) (*peerAuthenticationSenderFactory, error) {
-	return &peerAuthenticationSenderFactory{
-		messenger:                 args.messenger,
-		marshaller:                args.marshaller,
-		topic:                     args.topic,
-		timeBetweenSends:          args.timeBetweenSends,
-		timeBetweenSendsWhenError: args.timeBetweenSendsWhenError,
-		thresholdBetweenSends:     args.thresholdBetweenSends,
-		nodesCoordinator:          args.nodesCoordinator,
-		peerSignatureHandler:      args.peerSignatureHandler,
-		hardforkTrigger:           args.hardforkTrigger,
-		hardforkTimeBetweenSends:  args.hardforkTimeBetweenSends,
-		hardforkTriggerPubKey:     args.hardforkTriggerPubKey,
-		keysHolder:                args.keysHolder,
-		timeBetweenChecks:         args.timeBetweenChecks,
-		shardCoordinator:          args.shardCoordinator,
-		privKey:                   args.privKey,
-		redundancyHandler:         args.redundancyHandler,
-	}, nil
-}
-
-func (pasf *peerAuthenticationSenderFactory) create() (peerAuthenticationHandler, error) {
-	isMultikey := check.IfNil(pasf.privKey)
-	isRegular := check.IfNil(pasf.keysHolder)
-	if isMultikey && isRegular {
-		return nil, fmt.Errorf("%w for peerAuthenticationSenderFactory, received both validator key and multikey holder", heartbeat.ErrInvalidConfiguration)
+func createPeerAuthenticationSender(args argPeerAuthenticationSenderFactory) (peerAuthenticationSenderHandler, error) {
+	pk := args.privKey.GeneratePublic()
+	pkBytes, err := pk.ToByteArray()
+	if err != nil {
+		return nil, err
 	}
 
-	if !isMultikey && !isRegular {
-		return nil, fmt.Errorf("%w for peerAuthenticationSenderFactory, could not determine node's type", heartbeat.ErrInvalidConfiguration)
+	_, _, err = args.nodesCoordinator.GetValidatorWithPublicKey(pkBytes)
+	if err == nil {
+		return createRegularSender(args)
```

Comment:
please add a check here for  `len(keysMap)` and allow to create the regular peer only if that len is 0. Otherwise we should output error